File: programmers/greedy/hsj/체육복.py
def solution(n, lost, reserve):
    stu_list = [1 for i in range(n+1)]
    
    for j in lost:
        stu_list[j] -= 1
    
    for k in reserve:
        stu_list[k] += 1
    
    for m in range(1, n+1):
        if stu_list[m] == 2:
            if stu_list[m-1] == 0:
                stu_list[m] -= 1
                stu_list[m-1] += 1
            elif len(stu_list)-1 > m and stu_list[m+1] == 0:
                stu_list[m] -= 1
                stu_list[m+1] += 1
                
    v = len(stu_list)-stu_list.count(0)-1
    return v

File: programmers/greedy/체육복.js
function solution(n, lost, reserve) {
  var answer = 0;

  lost.sort((a, b) => a - b);

  for (let i = 0; i < lost.length; i++) {
    if (reserve.includes(lost[i])) {
      reserve.splice(reserve.indexOf(lost[i]), 1);
      lost.splice(i, 1);
      i--;
    }
  }
  for (let i = 0; i < lost.length; i++) {
    if (lost.length == 0) break;
    if (reserve.includes(lost[i] - 1) || reserve.includes(lost[i] + 1)) {
      if (reserve.includes(lost[i] - 1)) {
        reserve.splice(reserve.indexOf(lost[i] - 1), 1);
      } else {
        reserve.splice(reserve.indexOf(lost[i] + 1), 1);
      }
      lost.splice(i, 1);
      i--;
    }
  }
  answer = n - lost.length;

  return answer;
}
